src/OAuth2User.ts

```diff
@@ -39,13 +40,17 @@ export class OAuth2User implements OAuthClient {
   code_verifier?: string;
   code_challenge?: string;
   private _refreshAccessTokenPromise: Promise<{token: Token}> | null;
+  private _tokenEvents = new EventEmitter();
   constructor(options: OAuth2UserOptions) {
     const { token, ...defaultOptions } = options;
     this.options = {client_secret: '', ...defaultOptions};
     this.token = token;
     this._refreshAccessTokenPromise = null;
   }
-
+  
+  on(eventName: string, listener: (tokens: Token) => void) {
```

Comment:
Can you please type the return type. I don't think we should return this._tokenEvents; 

Could it be `: Promise<void>` and then change the below line to:
`await this._tokenEvents.on(eventName, listener);`

Reply:
resolved, but why we need await. after doing a promise<void>

Reply:
Right now the function won't wait for the .on async function to complete. Can you change it to one of these:
```
async on(eventName: EventName, listener: (tokens: Token) => void): Promise<void> {
    await this._tokenEvents.on(eventName, listener);
  }```
  
or (I think this one is better):
```
on(eventName: EventName, listener: (tokens: Token) => void): Promise<void> {
    return this._tokenEvents.on(eventName, listener);
  }
```

Reply:
but we need to return promise on  function inside on is complete. so we do need async and await.